Xamarin.Forms.ControlGallery.WindowsUniversal/SecondaryWindowService.cs

```diff
@@ -26,6 +26,10 @@ public async Task OpenSecondaryWindow(Type pageType)
 				ContentPage instance = (ContentPage)Activator.CreateInstance(pageType);
 				frame.Navigate(instance);
 				Window.Current.Content = frame;
+				Window.Current.SizeChanged += (sender, args) =>
```

Comment:
@KPixel makes a good point - why does this work? We should pull this out and create a separate issue.

Reply:
Ok i removed that code.. Also another code that works is something like this bellow. But we don't want to create a new Window right? 
```
public class NewSecondaryWindowsPage : WindowsPage
{
	protected override void OnNavigatedTo(Windows.UI.Xaml.Navigation.NavigationEventArgs e)
	{
		var pageType = e.Parameter as Type;
		ContentPage instance = (ContentPage)Activator.CreateInstance(pageType);
		RegisterWindow(instance);
		base.OnNavigatedTo(e);
	}
}

public async Task OpenSecondaryWindow(Type pageType)
{
	CoreApplicationView newView = CoreApplication.CreateNewView();
	int newViewId = 0;
	await newView.Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
	{
		var frame = new Windows.UI.Xaml.Controls.Frame();
		Window.Current.Content = frame;
		frame.Navigate(typeof(NewSecondaryWindowsPage), pageType);
		Window.Current.Activate();
		newViewId = ApplicationView.GetForCurrentView().Id;
	});
	bool viewShown = await ApplicationViewSwitcher.TryShowAsStandaloneAsync(newViewId);
}
```

Reply:
Interesting. I think this points to the root of the issue: [Application.MainPage](https://github.com/xamarin/Xamarin.Forms/blob/29b92aa1fc2b9d834bb3b95cc224e07d0cc3e9b4/Xamarin.Forms.Core/Application.cs#L66)...
I would like to create a new issue to discuss a solution there. Ok?

Reply:
Yeah the issue from what i understand is here :

https://github.com/xamarin/Xamarin.Forms/blob/719fc7a604ff0cce8922d717c99bfb0fa17e35e0/Xamarin.Forms.Platform.UAP/PageExtensions.cs#L58

We create a page just to fake it as "root" and so the code that listens to size changes here:

https://github.com/xamarin/Xamarin.Forms/blob/3ca1250330e8d22d4544aaa225301d42a8b5ec98/Xamarin.Forms.Platform.UAP/Platform.cs#L91

But the _page is not added to the VisualTree

Reply:
There is also the following code which expects all windows to have a specific structure:

https://github.com/xamarin/Xamarin.Forms/blob/3ca1250330e8d22d4544aaa225301d42a8b5ec98/Xamarin.Forms.Platform.UAP/Platform.cs#L51-L59

So, your code above seems like the expected way to initialize a secondary window.

I've updated my app to follow this approach, and it work.

Reply:
This means that the "fix" for the resize issue is to update `OpenSecondaryWindow()` in the Control Gallery app.

Also, the [PageExtensions](https://github.com/xamarin/Xamarin.Forms/blob/aa0b9c66f37b05f6c290a68aba39a0fc0e51aa3d/Xamarin.Forms.Platform.UAP/PageExtensions.cs#L44) class was a trap. From what I can see, in this entire repo, it is used in two scenarios:

1. For `FormsEmbeddedPageWrapper` pages (and there is no sample/test for that).
2. For the [UWP version of DualScreen](https://github.com/xamarin/Xamarin.Forms/blob/aa0b9c66f37b05f6c290a68aba39a0fc0e51aa3d/Xamarin.Forms.DualScreen/DualScreenHelper.uwp.cs#L43). And I think that code should be updated to follow the `WindowsPage` approach, otherwise, it will suffer from the same resize bug.